test/integration_test/basic_test.go

```diff
@@ -305,34 +313,107 @@ func BasicCsiRegression(tc *types.TestCase) func(*testing.T) {
-		require.NoError(t, err)
+		cluster = ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+		require.Equal(t, cluster.Spec.FeatureGates["CSI"], "false")
+
+		logrus.Info("Enable CSI and validate StorageCluster")
+		cluster.Spec.FeatureGates = map[string]string{"CSI": "true"}
+		cluster = ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+		require.Equal(t, cluster.Spec.FeatureGates["CSI"], "true")
+
+		// Delete and validate StorageCluster deletion
+		ci_utils.UninstallAndValidateStorageCluster(cluster, t)
+	}
+}
+
+// BasicStorkRegression test includes the following steps:
+// 1. Deploy PX with Stork enabled by default and validate Stork components and images
+// 2. Validate Stork is enabled by default
+// 3. Validate Stork webhook-controller is empty by default
+// 4. Validate Stork hostName is <nil> by default
+// 5. Delete "stork" pods and validate they get re-deployed
+// 6. Delete "stork-scheduler" pods and validate they get re-deployed
+// 7. Enable Stork webhook-controller and validate
+// 8. Disable Stork webhook-controller and validate
+// 9. Remove Stork webhook-controller and validate
+// 10. Enable hotNetwork and validate
+// 11. Disable hostNetwork and validate
+// 12. Remove hostNetwork and valiate
+// 13. Disable Stork and validate Stork components got successfully removed
+// 14. Enabled Stork and validate Stork components and images
+// 15. Delete StorageCluster and validate it got successfully removed
+func BasicStorkRegression(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		if tc.ShouldSkip() {
+			t.Skip()
+		}
+
+		var err error
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+
+		// Create and validate StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Validate Stork is enabled by default
+		require.True(t, cluster.Spec.Stork.Enabled, "failed to validate Stork is enabled by default, it should be enabled, but it is set to %v", cluster.Spec.Stork.Enabled)
 
-		// Sleep for 20 seconds to let operator start the update process
-		logrus.Debug("Sleeping for 20 seconds...")
-		time.Sleep(20 * time.Second)
+		// Validate Webhook controller arg doesn't exist by default
+		require.Empty(t, cluster.Spec.Stork.Args["webhook-controller"], "failed to validate webhook-controller, it shouldn't exist by default, but it is set to %s", cluster.Spec.Stork.Args["webhook-controller"])
 
-		logrus.Infof("Validate StorageCluster %s", cluster.Name)
-		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateUpdateTimeout, ci_utils.DefaultValidateUpdateRetryInterval, true, "")
+		// Validate HostNetwork is <nil> by default
+		require.Nil(t, cluster.Spec.Stork.HostNetwork, "failed to validate HostNetwork, it should be nil by default, but it isn't", cluster.Spec.Stork.HostNetwork)
```

Comment:
The last argument does not make sense as there is not print directive (%s) in message string.

Reply:
Added print of the cluster.Spec.Stork.HostNetwork value